cpp/src/arrow/csv/column_builder_test.cc

```diff
@@ -400,6 +400,24 @@ TEST_F(InferringColumnBuilderTest, MultipleChunkDate) {
                  ArrayFromJSON(date32(), "[null]")});
 }
 
+TEST_F(InferringColumnBuilderTest, SingleChunkTime) {
+  auto options = ConvertOptions::Defaults();
+  auto tg = TaskGroup::MakeSerial();
+
+  CheckInferred(tg, {{"", "01:23:45", "NA"}}, options,
```

Comment:
Could you include an integer or date (which should still allow the column to be inferred as time)?
```suggestion
  CheckInferred(tg, {{"", "99", "01:23:45", "NA"}}, options,
```

Reply:
I don't think so. Integers can not be parsed as a time field currently.

Reply:
I mean: I'm guessing that in the case of `"", "99", "01:23:45"` we'd fall back to string since we have an integer (definitely not a time, timestamp, ...) and a time (definitely not an integer, float, ...). That seems worth testing

Reply:
Will do.